### PR TITLE
writing TS functions

### PR DESCRIPTION
**Pull request by SeyedMojtaba1**
related #6

### CODE REVIEW

File: Seyed/TS syntax/.gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+node_modules/
+package.json
+package-lock.json
+tsconfig.json
```
Comment (fatemeh-ra) [inline Seyed/TS syntax/.gitignore:4]: you should NOT put `package.json`, `package-lock.json` and `tsconfig.json` in .gitignore

File: Seyed/TS syntax/syntax.ts [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+function power2(x: number) : bigint {
+    return BigInt(Math.pow(2,x));
+}
+
+
+// user object with first name and last name, both capitalized
+// e.g. getUser('hello', 'world') is { firstName: 'Hello', lastName: 'World' }
+// you should define the UserObject interface
+interface UserObject{
+    Fname : string,
```
Comment (fatemeh-ra) [inline Seyed/TS syntax/syntax.ts:35]: use camelCase format for variable names, it should be fName or firstName

File: Seyed/TS syntax/syntax.ts [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    Lname : string; 
+}
+
+function getUser(firstName: string, lastName: string): UserObject {
+    const x: UserObject = {
+        Fname : firstName,
```
Comment (fatemeh-ra) [inline Seyed/TS syntax/syntax.ts:41]: you should capitalize the first letter

File: Seyed/TS syntax/syntax.ts [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+}
+
+// higher order function: call the function in the parameters list
+// e.g. callFuncNTimes(flipACoin, 3) calls flipACoin 3 times
+function callFuncNTimes(func: () => string, n: number): void {
+    for(let i=0 ; i<n ; i++){
+        console.log(func());
+    }
+}
+// callFuncNTimes(flipACoin, 5);
```
Comment (fatemeh-ra) [inline Seyed/TS syntax/syntax.ts:93]: commented code are not allowed in the source code

File: Seyed/TS syntax/syntax.ts [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+
+// return true if parameter is falsy, false otherwise
+function isNil(param?: number): boolean {
```
Comment (fatemeh-ra) [inline Seyed/TS syntax/syntax.ts:97]: should return true with any falsy value including
```
0
-0
null
''
undefined
NaN
0n
false
```

File: Seyed/TS syntax/syntax.ts [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+    callableProp: function() {
+        console.log('hello!');
+    },
+    nonCallableProp: 42
+};
+
+
+// if the email is gmail
+function isGmail(email: string): boolean {
+    return email.includes("gmail");
```
Comment (fatemeh-ra) [inline Seyed/TS syntax/syntax.ts:133]: including is not enough, it should have the correct format: xx@gmail.com

File: Seyed/TS syntax/syntax.ts [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+}
+
+
+// duplicate array
+// e.g. duplicateArray([1,2,3]) is [1,2,3,1,2,3]
+function duplicateArray<T>(array1: Array<T>, array2: Array<T>): boolean {
```
Comment (fatemeh-ra) [inline Seyed/TS syntax/syntax.ts:139]: The issue description was not correct this function should have only one input array. and the return type was also incorrect.

**Review by fatemeh-ra — CHANGES_REQUESTED** [review]
Please fix the comments

File: Seyed/TS syntax/syntax.ts (hunk 2 above)
Comment (SeyedMojtaba1) [reply Seyed/TS syntax/syntax.ts]: I fixed it.

File: Seyed/TS syntax/syntax.ts (hunk 3 above)
Comment (SeyedMojtaba1) [reply Seyed/TS syntax/syntax.ts]: I did not understand it.

File: Seyed/TS syntax/syntax.ts (hunk 4 above)
Comment (SeyedMojtaba1) [reply Seyed/TS syntax/syntax.ts]: I deleted it.

File: Seyed/TS syntax/syntax.ts (hunk 5 above)
Comment (SeyedMojtaba1) [reply Seyed/TS syntax/syntax.ts:96]: I fixed it.

File: Seyed/TS syntax/syntax.ts (hunk 6 above)
Comment (SeyedMojtaba1) [reply Seyed/TS syntax/syntax.ts]: I fixed it.

File: Seyed/TS syntax/syntax.ts (hunk 7 above)
Comment (SeyedMojtaba1) [reply Seyed/TS syntax/syntax.ts]: I fixed it.

**Review by fatemeh-ra — APPROVED** [review]

**Pull request merged**